File: logs/glue/commons-fileupload/src/main/java/org/apache/commons/fileupload/disk/DiskFileItemFactory.java
package org.apache.commons.fileupload.disk;

import java.io.File;
import org.apache.commons.fileupload.ContextInitializer;
import org.apache.commons.fileupload.ExceptionHandler;
import org.apache.commons.fileupload.IntegrationUtils;
import org.graalvm.polyglot.PolyglotException;
import org.graalvm.polyglot.Value;

public class DiskFileItemFactory {
  private transient Value obj =
      IntegrationUtils.createDefaultPythonObject(
          ContextInitializer.getPythonClass(
              "main/org/apache/commons/fileupload/disk/DiskFileItemFactory.py",
              "DiskFileItemFactory"));
  public static final int DEFAULT_SIZE_THRESHOLD = 10240;
  public File repository;
  public int sizeThreshold = DEFAULT_SIZE_THRESHOLD;
  public String defaultCharset = DiskFileItem.DEFAULT_CHARSET;

  public Value getPythonObject() {
    return obj;
  }

  public void setPythonObject(Value obj) {
    this.obj = obj;
  }

  public void setDefaultCharset(String pCharset) {

    defaultCharset = pCharset;
  }

  public String getDefaultCharset() {

    return defaultCharset;
  }

  public void setSizeThreshold(int sizeThreshold) {

    this.sizeThreshold = sizeThreshold;
  }

  public int getSizeThreshold() {

    return sizeThreshold;
  }

  public void setRepository(File repository) {

    this.repository = repository;
  }

  public File getRepository() {

    return repository;
  }

  public static DiskFileItemFactory DiskFileItemFactory1() {

    return new DiskFileItemFactory(DEFAULT_SIZE_THRESHOLD, null);
  }

  public DiskFileItemFactory(int sizeThreshold, File repository) {

    this.sizeThreshold = sizeThreshold;
    this.repository = repository;
  }

  public java.util.Map pyToJ() {
    java.util.Map idMap = new java.util.HashMap();
    return pyToJ(idMap);
  }

  public java.util.Map pyToJ(java.util.Map idMap) {

    this.repository =
        (File)
            IntegrationUtils.valueToObject(
                this.obj.getMember("_DiskFileItemFactory__repository"), "File", idMap, repository);
    this.sizeThreshold =
        (int)
            IntegrationUtils.valueToObject(
                this.obj.getMember("_DiskFileItemFactory__sizeThreshold"),
                "int",
                idMap,
                sizeThreshold);
    this.defaultCharset =
        (String)
            IntegrationUtils.valueToObject(
                this.obj.getMember("_DiskFileItemFactory__defaultCharset"),
                "String",
                idMap,
                defaultCharset);
    return idMap;
  }

  public Value jToPy() {
    Value idMap = IntegrationUtils.mapToPython(new java.util.HashMap());
    return jToPy(idMap);
  }

  public Value jToPy(Value idMap) {

    this.obj.invokeMember("__setattr__", "javaObj", this);
    this.obj.invokeMember(
        "__setattr__",
        "_DiskFileItemFactory__repository",
        IntegrationUtils.mapToPython(
            repository, idMap, this.obj.getMember("_DiskFileItemFactory__repository")));
    this.obj.invokeMember(
        "__setattr__",
        "_DiskFileItemFactory__sizeThreshold",
        IntegrationUtils.mapToPython(
            sizeThreshold, idMap, this.obj.getMember("_DiskFileItemFactory__sizeThreshold")));
    this.obj.invokeMember(
        "__setattr__",
        "_DiskFileItemFactory__defaultCharset",
        IntegrationUtils.mapToPython(
            defaultCharset, idMap, this.obj.getMember("_DiskFileItemFactory__defaultCharset")));
    return idMap;
  }
}

File: data/translated_projects/commons-fileupload/src/main/org/apache/commons/fileupload/disk/DiskFileItemFactory.py
from __future__ import annotations
import pathlib
import io
import os
from src.main.org.apache.commons.fileupload.disk.DiskFileItem import *
import java # type: ignore
from src.main.org.apache.commons.fileupload.java_handler import *
class DiskFileItemFactory(metaclass=StaticFieldRedirector):
    def setDefaultCharset(self, pCharset: str) -> None:

        idMapPyToJ = self.javaObj.pyToJ()
        translatedArg0 = JavaHandler.valueToObject(pCharset, "String", idMapPyToJ)
        idMapJToPy = dict()
        try:
            self.javaObj.setDefaultCharset(translatedArg0)
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            self.javaObj.jToPy(idMapJToPy)
            JavaHandler.mapping(translatedArg0, idMapJToPy, pCharset)
    def getDefaultCharset(self) -> str:

        idMapPyToJ = self.javaObj.pyToJ()
        
        idMapJToPy = dict()
        try:
            val = JavaHandler.mapping(self.javaObj.getDefaultCharset(), id_map=idMapJToPy)
            return val
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            self.javaObj.jToPy(idMapJToPy)
            
    def setSizeThreshold(self, sizeThreshold: int) -> None:

        idMapPyToJ = self.javaObj.pyToJ()
        translatedArg0 = JavaHandler.valueToObject(sizeThreshold, "int", idMapPyToJ)
        idMapJToPy = dict()
        try:
            self.javaObj.setSizeThreshold(translatedArg0)
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            self.javaObj.jToPy(idMapJToPy)
            JavaHandler.mapping(translatedArg0, idMapJToPy, sizeThreshold)
    def getSizeThreshold(self) -> int:

        idMapPyToJ = self.javaObj.pyToJ()
        
        idMapJToPy = dict()
        try:
            val = JavaHandler.mapping(self.javaObj.getSizeThreshold(), id_map=idMapJToPy)
            return val
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            self.javaObj.jToPy(idMapJToPy)
            
    def setRepository(self, repository: pathlib.Path) -> None:

        idMapPyToJ = self.javaObj.pyToJ()
        translatedArg0 = JavaHandler.valueToObject(repository, "File", idMapPyToJ)
        idMapJToPy = dict()
        try:
            self.javaObj.setRepository(translatedArg0)
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            self.javaObj.jToPy(idMapJToPy)
            JavaHandler.mapping(translatedArg0, idMapJToPy, repository)
    def getRepository(self) -> pathlib.Path:

        idMapPyToJ = self.javaObj.pyToJ()
        
        idMapJToPy = dict()
        try:
            val = JavaHandler.mapping(self.javaObj.getRepository(), id_map=idMapJToPy)
            return val
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            self.javaObj.jToPy(idMapJToPy)
            
    @staticmethod

    def DiskFileItemFactory1() -> DiskFileItemFactory:

        idMapPyToJ = JavaHandler.valueToObject(dict(), "Map")
        
        idMapJToPy = dict()
        try:
            val = JavaHandler.mapping(DiskFileItemFactory.javaClz.DiskFileItemFactory1(), id_map=idMapJToPy)
            return val
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass

            
    def __init__(self, sizeThreshold: int, repository: pathlib.Path) -> None:

        idMapPyToJ = JavaHandler.valueToObject(dict(), "Map")
        translatedArg0 = JavaHandler.valueToObject(sizeThreshold, "int", idMapPyToJ)
        translatedArg1 = JavaHandler.valueToObject(repository, "File", idMapPyToJ)
        idMapJToPy = dict()
        try:
            self.javaObj = DiskFileItemFactory.javaClz(translatedArg0, translatedArg1)
            self.javaObj.setPythonObject(self)
            self.javaObj.jToPy(idMapJToPy)
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            JavaHandler.mapping(translatedArg0, idMapJToPy, sizeThreshold)
            JavaHandler.mapping(translatedArg1, idMapJToPy, repository)
    javaClz = java.type("org.apache.commons.fileupload.disk.DiskFileItemFactory")
    def __getattr__(self, name):
        try:
            return object.__getattribute__(self, name)
        except AttributeError:
            pass
        return JavaHandler.mapping(getattr(self.javaClz, StaticFieldRedirector.unmangle_name(name)))